addons/l10n_ar/models/account_invoice.py

```diff
@@ -0,0 +1,473 @@
+
+            # vat exempt values (are the ones with code 2)
+            vat_exempt_taxes = rec.tax_line_ids.filtered(
+                lambda r: (
+                    r.tax_id.tax_group_id.l10n_ar_type == 'tax' and
+                    r.tax_id.tax_group_id.l10n_ar_tax == 'vat' and
+                    r.tax_id.tax_group_id.l10n_ar_afip_code == 2))
+            rec.l10n_ar_vat_exempt_base_amount = sum(
+                vat_exempt_taxes.mapped('base'))
+
+            # vat untaxed values / no gravado (are the ones with code 1)
+            vat_untaxed_taxes = rec.tax_line_ids.filtered(
+                lambda r: (
+                    r.tax_id.tax_group_id.l10n_ar_type == 'tax' and
+                    r.tax_id.tax_group_id.l10n_ar_tax == 'vat' and
+                    r.tax_id.tax_group_id.l10n_ar_afip_code == 1))
+            rec.l10n_ar_vat_untaxed_base_amount = sum(
+                vat_untaxed_taxes.mapped('base'))
+
+            # other taxes values
+            not_vat_taxes = rec.tax_line_ids - vat_taxes
+            l10n_ar_other_taxes_amount = sum(not_vat_taxes.mapped('amount'))
+            rec.l10n_ar_not_vat_tax_ids = not_vat_taxes
+            rec.l10n_ar_other_taxes_amount = l10n_ar_other_taxes_amount
+
+    @api.multi
+    @api.depends('l10n_latam_document_number', 'number')
+    def _compute_l10n_ar_invoice_number(self):
+        for rec in self.filtered('l10n_latam_document_number'):
+            str_number = rec.l10n_latam_document_number
+            if rec.l10n_latam_document_type_id.code in ['33', '99', '331', '332']:
+                point_of_sale = '0'
+                # leave only numbers and convert to integer
+                # otherwise use date as a number
+                if re.search(r'\d', str_number):
+                    invoice_number = str_number
+                else:
+                    invoice_number = rec.date_invoice
+            # despachos de importacion
+            elif rec.l10n_latam_document_type_id.code == '66':
+                point_of_sale = '0'
+                invoice_number = '0'
+            elif "-" in str_number:
+                splited_number = str_number.split('-')
+                invoice_number = splited_number.pop()
+                point_of_sale = splited_number.pop()
+            elif "-" not in str_number and len(str_number) == 12:
+                point_of_sale = str_number[:4]
+                invoice_number = str_number[-8:]
+            else:
+                raise ValidationError(_(
+                    'Could not get invoice number and point of sale for '
+                    'Invoice [%i] %s') % (rec.id, rec.display_name))
+            rec.l10n_ar_invoice_number = int(
+                re.sub("[^0-9]", "", invoice_number))
+            rec.l10n_ar_afip_pos_number = int(
+                re.sub("[^0-9]", "", point_of_sale))
+
+    @api.multi
+    def get_localization_invoice_vals(self):
+        self.ensure_one()
+        if self.company_id.country_id.code == 'AR':
```

Comment:
```suggestion
        if self.company_id.country_id.code != 'AR':
            return super(
                AccountInvoice, self).get_localization_invoice_vals()
```
With this, you avoid the next else, and an interation